src/answers-headless.ts

```diff
@@ -27,108 +27,247 @@ import answersUtilities from './answers-utilities';
 import { SelectableFilter } from './models/utils/selectablefilter';
 import { transformFiltersToCoreFormat } from './utils/transform-filters';
 
+/**
+ * Provides methods for updating the state and for sending requests to the Answers
+ * API through {@link AnswersCore}.
+ */
 export default class AnswersHeadless {
+  /**
+   * Common utility functions for manipulating Answers-related data.
+   */
   public readonly utilities = answersUtilities;
 
+  /**
+   * Sets up the AnswersHeadless instance with managers for its state and network
+   * requests and an instance of the {@link AnswersCore} library.
+   *
+   * @param core - The Answers Core instance used for sending network requests
+   * @param stateManager - The manager of the state and how it can be manipulated
+   * @param httpManager - The manager for network requests and reponses through core
+   */
   constructor(
     private core: AnswersCore,
     private stateManager: StateManager,
     private httpManager: HttpManager,
   ) {}
 
+  /**
+   * Sets {@link QueryState.input} to the specified input.
+   *
+   * @param input - The new query input to update to
```

Comment:
What do you think about the wording "The input which is set" instead?

Reply:
Sure, I can change it to "The input to set" (and similar for the others)